src/librustdoc/json/mod.rs

```diff
@@ -189,6 +189,11 @@ impl<'tcx> FormatRenderer<'tcx> for JsonRenderer<'tcx> {
 
     fn after_krate(&mut self) -> Result<(), Error> {
         debug!("Done with crate");
+
+        for primitive in Rc::clone(&self.cache).primitive_locations.values() {
+            self.get_impls(primitive.clone());
+        }
```

Comment:
Can you add a test for this change? There are examples in `src/test/rustdoc-json`.